bazelw

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+set -euo pipefail
+
+readonly bazelisk_version="1.10.1"
+if [[ $OSTYPE == darwin* ]]; then
+  # TODO: Support M1 once https://github.com/envoyproxy/envoy/issues/16482
+  readonly bazel_platform="darwin-amd64"
+  readonly bazel_version_sha="e485bbf84532d02a60b0eb23c702610b5408df3a199087a4f2b5e0995bbf2d5a"
+else
+  readonly bazel_platform="linux-amd64"
+  readonly bazel_version_sha="4cb534c52cdd47a6223d4596d530e7c9c785438ab3b0a49ff347e991c210b2cd"
+fi
+
+readonly bazel_version_url="https://github.com/bazelbuild/bazelisk/releases/download/v$bazelisk_version/bazelisk-$bazel_platform"
+script_root="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+readonly bazelisk="$script_root/tmp/bazel/versions/bazelisk-$bazelisk_version-$bazel_platform"
+
+if [[ ! -x "$bazelisk" ]]; then
+  echo "Installing bazelisk..." >&2
+  mkdir -p "$(dirname "$bazelisk")"
+
+  download_bazelisk() {
+    curl --fail -L --retry 5 --retry-connrefused --silent --progress-bar \
+      --output "$bazelisk" "$bazel_version_url"
+  }
+
+  download_bazelisk || download_bazelisk
```

Comment:
Why retry this on failure when we already have --retry 5 above?

Reply:
hrm yea, I kinda copied this from our internal script, I'm not sure if one predated the other, or if this attempted to retry a failure that curl didn't retry as intended

Reply:
Can't hurt too much.